light-client/examples/light_client.rs

```diff
@@ -109,20 +103,11 @@ fn sync_cmd(opts: SyncOpts) {
     let verifier = ProdVerifier::default();
     let clock = SystemClock;
     let scheduler = scheduler::basic_bisecting_schedule;
-    let fork_detector = ProdForkDetector::default();
-
-    let mut light_client = LightClient::new(
-        state,
-        options,
-        clock,
-        scheduler,
-        verifier,
-        fork_detector,
-        io,
-    );
+
+    let mut light_client = LightClient::new(primary, options, clock, scheduler, verifier, io);
 
     loop {
-        match light_client.verify_to_highest() {
+        match light_client.verify_to_highest(&mut state) {
```

Comment:
If the state is `Instance`  specific would it make sense to just make it internal and have the `Instance` expose specific methods for updating the latest trusted state? This might provide better encapsulation. 🤔